### PR TITLE
Story feature complete flow

### PR DESCRIPTION
**Pull request by titor-oopart**

### CODE REVIEW

File: src/main/java/pivotaltracker/PageTransporter.java [hunk 1]
```diff
@@ -48,6 +48,11 @@ public static void navigatePage(final String keyUrl) {
         goToURL(baseUrl.concat(keyUrl));
     }
 
+    public static void navigatePageThroughId(String keyUrl, int id) {
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/PageTransporter.java]: this method need documentation

File: src/main/java/pivotaltracker/ProjectObject.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ *
+ * @author John Salazar Pinto
+ * @version 1.0
+ */
+public class ProjectObject {
+    private static int idProject;
+    private int storyPoints;
+    private String projectName;
+
+    public ProjectObject(){
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/ProjectObject.java]: this method need documentation

File: src/main/java/pivotaltracker/ProjectObject.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    public String getProjectName() {
+        return projectName;
+    }
+
+    /**
+     * Sets project name.
+     *
+     * @param projectName
+     */
+    public void setProjectName(String projectName) {
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/ProjectObject.java]: You used the checkstyle. Looks like you didn't.

File: src/main/java/pivotaltracker/StoryObject.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    public int getProject_id() {
+        return project_id;
+    }
+
+    /**
+     * Sets project id.
+     *
+     * @param project_id
+     */
+    public void setProject_id(int project_id) {
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/StoryObject.java]: You used the checkstyle. Looks like you didn't.

File: src/main/java/pivotaltracker/entities/Context.java [hunk 5]
```diff
@@ -40,6 +44,19 @@ public Account getAccount() {
...
+    /**
+     * Returns the project of context.
+     *
+     * @return project
+     */
+    public Project getProject() {
+        return project;
+    }
+
+    public Story getStory() {
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/entities/Context.java:64]: this method need documentation

File: src/main/java/core/utils/CSVReader.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            while (br.readLine() != null) {
+                linenumber++;
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        System.out.println(linenumber + " This is the line number");
+        String storyData[] = new String[linenumber];
+        try {
+            br = new BufferedReader(new FileReader(csvFile));
```
Comment (Cxristian) [inline src/main/java/core/utils/CSVReader.java]: what is br?change name

File: src/main/java/pivotaltracker/PageTransporter.java [hunk 7]
```diff
@@ -48,10 +48,34 @@ public static void navigatePage(final String keyUrl) {
...
+        goToURL(baseUrl.concat(keyUrl).concat("/").concat(String.valueOf(id)));
+    }
+
+    /**
+     * Navigates on page with id.
+     *
+     * @param keyUrl - Url key.
+     * @param id     - Project id.
+     */
+    public static void navigatePageThroughId(final String keyUrl, final int id, final String endpoint) {
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/PageTransporter.java]: This method is the same as the method called: navigatePageById

File: src/main/java/pivotaltracker/entities/Project.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
+package pivotaltracker.entities;
+
+/**
+ * Project class.
+ *
+ * @author John Salazar Pinto
+ * @version 1.0
+ */
+public class Project {
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/entities/Project.java]: Do you use this class?

File: src/main/java/pivotaltracker/ui/pages/story/StoryPage.java [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+
+    private static final String ADDSTORYBTN = "//div[@id='panel_backlog_%S'] //a[@title='Add Story']";
+    private static final String BACKLOGFOCUS = "panel_backlog_%S";
+    private static final String STORYTYPE = "//a[@class='item_%s ']";
+    private static final String STARTSTORY = "//div[@aria-label='%s'] //label[@data-aid='StateButton']";
+    private static final String ACCEPTSTORY = "//div[@aria-label='%s'] //label[text()='Accept']";
+    private static final String ESTIMATESTORY = "//div[@data-id='%s'] //span[@class='meta'] //span";
+
+    @Override
+    protected void waitUntilPageObjectIsLoaded() {
```
Comment (LimbertVargas) [inline src/main/java/pivotaltracker/ui/pages/story/StoryPage.java:97]: maybe you can write code in this method

File: src/main/java/pivotaltracker/ui/pages/story/ImportProjectPage.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    private void clickImportBtn() {
+        importBtn.click();
+    }
+
+    /**
+     * Sends path for upload file.
+     */
+    private void sendPath() {
+        DriverMethods.setText(chooseFileBtn, System.getProperty("user.dir") + "/" + "src/test/resources/files/test_20191006_1736.csv");
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/ui/pages/story/ImportProjectPage.java]: change your system.getPtoperty in a constant

File: src/main/java/pivotaltracker/ui/pages/story/StoryPage.java [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+    private WebElement labelTxt;
+
+    private static final String ADDSTORYBTN = "//div[@id='panel_backlog_%S'] //a[@title='Add Story']";
+    private static final String BACKLOGFOCUS = "panel_backlog_%S";
+    private static final String STORYTYPE = "//a[@class='item_%s ']";
+    private static final String STARTSTORY = "//div[@aria-label='%s'] //label[@data-aid='StateButton']";
+    private static final String ACCEPTSTORY = "//div[@aria-label='%s'] //label[text()='Accept']";
+    private static final String ESTIMATESTORY = "//div[@data-id='%s'] //span[@class='meta'] //span";
+
+    @Override
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/ui/pages/story/StoryPage.java:96]: add documentation and wait until

File: src/main/java/pivotaltracker/ui/pages/story/StoryPage.java [hunk 12]
```diff
@@ -0,0 +1,274 @@
...
+    private WebElement labelTxtBox;
+
+    @FindBy(css = "[class='autosaves button std save']")
+    private WebElement saveStoryBtn;
+
+    @FindBy(xpath = "//span[@class='meta'] //span")
+    private WebElement estimateTxt;
+
+    @FindBy(css = "[class='tracker_markup']")
+    private WebElement storyName;
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/ui/pages/story/StoryPage.java]: put your type of web element

File: src/main/java/pivotaltracker/ui/pages/story/StoryPage.java [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+    public void createBacklogStory(final String storyTitle) {
+        DriverMethods.setText(storyTitleTxtBox, storyTitle);
+    }
+
+    /**
+     * Gets id project.
+     *
+     * @return idProject
+     */
+    private String getIdProject() {
```
Comment (Cxristian) [inline src/main/java/pivotaltracker/ui/pages/story/StoryPage.java:123]: what does this the method?

File: src/test/java/steps/ImportExportProject.java [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+        PageTransporter.navigatePageThroughId(endpoint, ProjectObject.getIdProject(), "export");
+    }
+
+    @When("I export the stories in a CSV File")
+    public void iExportTheStoriesInACSVFile() {
+        System.setProperty("webdriver.chrome.driver", System.getProperty("user.dir"));
+        exportProjectPage.exportProject();
+    }
+
+    @Then("I should see the file in the folder download")
```
Comment (Cxristian) [inline src/test/java/steps/ImportExportProject.java]: add documentation and the name method must start with a verb

File: src/test/java/steps/StorySteps.java [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    /**
+     * Asserts the data of the story.
+     */
+    @Then("I should see the story on Accepted Stories on Current Iteration panel")
+    public void iShouldSeeTheStoryOnAcceptedStoriesOnCurrentIterationPanel() {
+        StoryAPI storyAPI = new StoryAPI();
+        storyAPI.getStories();
+        System.out.println(context.getStoryObject().getName() + "asdf");
```
Comment (Cxristian) [inline src/test/java/steps/StorySteps.java]: delete souts...

File: src/test/java/steps/StorySteps.java [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+     * Constructor class.
+     *
+     * @param context
+     */
+    public StorySteps(Context context) {
+        this.context = context;
+        story = context.getStory();
+    }
+
+    @When("I go to the Story page")
```
Comment (Cxristian) [inline src/test/java/steps/StorySteps.java:49]: add documents

File: src/test/java/steps/StorySteps.java [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    @And("I finish and deliver the story")
+    public void iFinishAndDeliverTheStory() {
+        storyPage.finishStoryFlow(story.getStoryName());
+    }
+
+    /**
+     * Asserts the data of the story.
+     */
+    @Then("I should see the story on Accepted Stories on Current Iteration panel")
```
Comment (Cxristian) [inline src/test/java/steps/StorySteps.java]: change the name method

File: src/main/java/pivotaltracker/PageTransporter.java (hunk 1 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/PageTransporter.java]: done

File: src/main/java/pivotaltracker/ProjectObject.java (hunk 2 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/ProjectObject.java]: done

File: src/main/java/pivotaltracker/ProjectObject.java (hunk 3 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/ProjectObject.java]: done

File: src/main/java/pivotaltracker/StoryObject.java (hunk 4 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/StoryObject.java]: done

File: src/main/java/pivotaltracker/entities/Context.java (hunk 5 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/entities/Context.java:64]: done

File: src/main/java/core/utils/CSVReader.java (hunk 6 above)
Comment (titor-oopart) [reply src/main/java/core/utils/CSVReader.java]: done

File: src/main/java/pivotaltracker/PageTransporter.java (hunk 7 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/PageTransporter.java]: done

File: src/main/java/pivotaltracker/entities/Project.java (hunk 8 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/entities/Project.java]: No

File: src/main/java/pivotaltracker/ui/pages/story/ImportProjectPage.java (hunk 10 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/ui/pages/story/ImportProjectPage.java]: Done

File: src/main/java/pivotaltracker/ui/pages/story/StoryPage.java (hunk 11 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/ui/pages/story/StoryPage.java:96]: done

File: src/main/java/pivotaltracker/ui/pages/story/StoryPage.java (hunk 12 above)
Comment (titor-oopart) [reply src/main/java/pivotaltracker/ui/pages/story/StoryPage.java]: done

**Pull request closed without merging**